apisix/plugins/auth-casdoor.lua

```diff
@@ -0,0 +1,122 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local cjson = require("cjson")
+local http = require("resty.http")
+local session = require("resty.session")
+local log = core.log
+
+local plugin_name = "auth-casdoor"
```

Comment:
Should be authz-casdoor like authz-keycloak and so on?

Reply:
Well I was given clear instruction that the name of this plugin should be this. It is because that authz means authorization, but we are actually doing is mostly (authn) authentication, but authorization is also involved. Therefore maybe auth is better.

Reply:
This plugin implements a variant of oauth2 which is for authorization, just like authz-keycloak.

Reply:
> but we are actually doing is mostly (authn) authentication

No, but we are actually doing is mostly (authz) authorization.

1. OAuth 2.0 is an authorization protocol and NOT an authentication protocol. 
2. the endpoints in this plugin also indicates that: "/api/login/oauth/access_token" and "/login/oauth/authorize".

Reply:
Well perhaps I ... prefer to let the  plugin's name remain as it is now, and I wonder whether it is ok.